extra_data/reader.py

```diff
@@ -278,8 +278,13 @@ def __getitem__(self, item):
             return self._get_key_data(*item)
         elif isinstance(item, str):
             return self._get_source_data(item)
+        elif (
+            isinstance(item, (by_id, by_index, list, np.ndarray, slice)) or
+            is_int_like(item)
+        ):
+            return self.select_trains(item)
 
-        raise TypeError("Expected data[source] or data[source, key]")
+        raise TypeError("Expected data[source], data[source, key] or data[train_selection]")
```

Comment:
While trying to trigger this error, I ran across a surprising (to me) behavior, the following does not raise this error:
```python
run['SA3_XTD10_XGM/XGM/DOOCS:output', 2]
```
rather, it returns a very sensible object:
```python
<extra_data.SourceData source='SA3_XTD10_XGM/XGM/DOOCS:output' for 1 trains>
```

I suppose this is part of `train_selection`  but I couldn't easily find it documented anywhere.

Reply:
huh, interesting side effect. but yes, since this essentially does `run[source][key]` and now that `SourceData[key]` can be used for filtering trains, it makes sense.
I'm fine keeping that behavior, unless others object.

Reply:
Mhh, I *think* I would error on the side of caution here and raise an exception:
* Supporting `DataCollection[source, train_sel]` begs the question why not `DataCollection[source, key, train_sel]`
* It would hide (and make it really hard to debug) type errors in your code, say you call `run[source, key]` and due to some other bug `key` becomes an `int` - a wrong `str` would raise an exception, but a random `int` doesn't?

Reply:
Well, as long as the random `int` is larger than zero and less than `len(run.train_ids)` but that's a large range, typically, and that's what you mean?